pkg/operator/operator.go

```diff
@@ -39,10 +40,17 @@ type Operator struct {
 	StatusManager *statusmanager.StatusManager
 }
 
+var logger = klog.NewKlogr()
+
 func RunOperator(ctx context.Context, controllerConfig *controllercmd.ControllerContext, inClusterClientName string, extraClusters map[string]string) error {
 	o := &Operator{}
 
 	var err error
+
+	// Call SetLogger before adding controller-runtime client to prevent controller-runtime
+	// complaining about it after 30 seconds of binaries lifetime
+	// https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/log/log.go#L54
+	ctlog.SetLogger(logger)
```

Comment:
hmm isn't `RunOperator` only called once from main?

Reply:
do you think
```
func main() {
	pflag.CommandLine.SetNormalizeFunc(utilflag.WordSepNormalizeFunc)
	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)

	logs.InitLogs()
	defer logs.FlushLogs()

	command := newNetworkOperatorCommand()

	if err := command.Execute(); err != nil {
		fmt.Fprintf(os.Stderr, "%v\n", err)
		os.Exit(1)
	}
}
```
is a better spot?

Reply:
I confused the functions, apologies.
In that case I am fine to keep this in RunOperator as this is the main entry-point of CNO anyway.

Reply:
Yes, RunOperator only gets called once from main. I was little bit worried about hard coded condition of initializing SetLogger within 30 second of binary's lifetime and thats why I have put it in RunOperator instead of main.

https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/log/log.go#L54

I have also tested by initializing the logger just before command.Execute() and that would also resolve the panic.